src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -1455,6 +1455,159 @@ public static void Test()
             }
         }
 
+        [Fact]
+        public static void BigInteger_Floor()
```

Comment:
If these were `[Theory]`s that took pairs of values then the message produced would be clearer as to the failing input if a test failed, making fixing it easier.

Reply:
This code generates the same `AssertActualExpectedException` that would be generated when using a `[Theory]`, but it runs a lot faster. `GetRandomData` produces about N=1000 decimals that are then tested in N * N, N * 29 or N invocations depending on the test.